okhttp/src/main/java/okhttp3/OkHttpTrustManager.kt

```diff
@@ -0,0 +1,159 @@
+package okhttp3
+
+import android.annotation.SuppressLint
+import java.lang.reflect.InvocationTargetException
+import java.lang.reflect.Method
+import java.net.Socket
+import java.security.cert.Certificate
+import java.security.cert.X509Certificate
+import javax.net.ssl.SSLEngine
+import javax.net.ssl.X509ExtendedTrustManager
+import javax.net.ssl.X509TrustManager
+import okhttp3.internal.platform.Platform
+import org.codehaus.mojo.animal_sniffer.IgnoreJRERequirement
+
+interface OkHttpTrustManager : X509TrustManager {
```

Comment:
@swankjesse I'm not sold on this.  It's making the public API we wanted to avoid.  This draft is musing on the right API if we want to support it.  Not sure why else you would want to override the TrustManager?

It's probably to have two distinct TrustManagers e.g. from different keystores?

So maybe it's a selecting implementation and we provide a utility like OkHttpTrustManager.INSECURE that would only be used for the devserver.

Reply:
Do people want to mix a secure and an insecure trust manager? I suspect an `INSECURE_TRUST_MANAGER` constant might be the easiest way to get up and running quickly?

Perhaps even something that doesn’t expose the TrustManager type:

```
val client = OkHttpClient.Builder()
     .insecureTrustManager("🛑 NOT FOR PRODUCTION USE 🛑")
     .build()
```

In the above example we’d require that literal string to be passed in; any other string would throw.

Reply:
Give me one more go :) before we get to magic strings...

Reply:
Do we need a new interface for this thing? Seems like the existing TrustManager interface is sufficient!